src/change_stream/mod.rs

```diff
@@ -0,0 +1,160 @@
+/// #       eprintln!("Failed command: {:?}", event);
+/// #   }
+/// # }
+///
+/// # async fn do_stuff() -> Result<()> {
+/// # let handler: Arc<dyn CommandEventHandler> = Arc::new(FailedCommandLogger);
+/// # let options = ClientOptions::builder()
+/// #                 .command_event_handler(handler)
+/// #                 .build();
+/// # let client = Client::with_options(options)?;
+/// # let coll = client.database("foo").collection("bar");
+/// let mut change_stream = coll.watch(None, None).await?;
+/// while let Some(change) = change_stream.next().await {
+///     println!("{:?}", change?)
+/// }
+///
+/// # Ok(())
+/// # }
+/// ```
+///
+/// See the documentation [here](https://docs.mongodb.com/manual/changeStreams) for more
+/// details. Also see the documentation on [usage recommendations](https://docs.mongodb.com/manual/administration/change-streams-production-recommendations/).
+pub struct ChangeStream<'a, T: Deserialize<'a> = ChangeStreamEventDocument> {
+    /// The cursor to iterate over `ChangeStreamEventDocument` instances.
+    cursor: Cursor,
+
+    /// The pipeline of stages to append to an initial `$changeStream` stage.
+    pipeline: Vec<Document>,
+
+    /// The client that was used for the initial `$changeStream` aggregation, used for server
+    /// selection during an automatic resume.
+    client: Client,
+
+    /// The original target of the change stream, used for re-issuing the aggregation during
+    /// an automatic resume.
+    target: ChangeStreamTarget,
+
+    /// The cached resume token.
+    resume_token: Option<ResumeToken>,
+
+    /// The options provided to the initial `$changeStream` stage.
+    options: Option<ChangeStreamOptions>,
+
+    /// The read preference for the initial `$changeStream` aggregation, used for server selection
+    /// during an automatic resume.
+    read_preference: Option<ReadPreference>,
+
+    /// Whether or not the change stream has attempted a resume, used to attempt a resume only
+    /// once.
+    resume_attempted: bool,
+
+    /// Whether or not the change stream has returned a document, used to update resume token
+    /// during an automatic resume.
+    document_returned: bool,
+
+    /// Necessary to satisfy the compiler that `T` and `'a` are used in the struct definition.
+    ///
+    /// See the [Rust docs](https://doc.rust-lang.org/std/marker/struct.PhantomData.html) for more details.
+    phantom: PhantomData<&'a T>,
+}
+
+impl<'a, T> ChangeStream<'a, T>
+where
+    T: Deserialize<'a>,
+{
+    /// Returns the cached resume token that can be used to resume after the most recently returned
+    /// change.
+    ///
+    /// See the documentation
+    /// [here](https://docs.mongodb.com/manual/changeStreams/#change-stream-resume-token) for more
+    /// information on change stream resume tokens.
+    pub fn resume_token(&self) -> Option<ResumeToken> {
+        todo!();
+    }
+
+    fn resume(&mut self) -> Result<()> {
+        todo!();
```

Comment:
`todo!()` is an expression, so semicolons aren't necessary when using it. No need to update these though, since we'll remove all of them before merging into master

Reply:
got it